src/google-classroom-extractor/edfi_google_classroom_extractor/mapping/assignment_submissions.py

```diff
@@ -4,10 +4,63 @@
 # See the LICENSE and NOTICES files in the project root for more information.
 
 from typing import Dict, Tuple, Any
-from pandas import DataFrame
+from pandas import DataFrame, isna
 from edfi_google_classroom_extractor.mapping.constants import SOURCE_SYSTEM
 
+# States returned by API
+CREATED_STATE = "CREATED"
+NEW_STATE = "NEW"
+RECLAIMED_STATE = "RECLAIMED_BY_STUDENT"
 TURNED_IN_STATE = "TURNED_IN"
+RETURNED_STATE = "RETURNED"
+
+# States derived from API "late" flag
+LATE_STATE = "LATE"
+MISSING_STATE = "MISSING"
+
+
+def derive_state(submission_row: Any) -> str:
```

Comment:
Looks like `submission_row` is at least a `dict` rather than any `Any`...

Reply:
No, I see now that this is not from `iterrows` but from `apply`. So the type should be `pandas.core.series.Series`, I think.

Reply:
Nice catch, Any was a placeholder. Should be pandas Series.